zebra-state/src/service.rs

```diff
@@ -1288,6 +1289,31 @@ impl Service<ReadRequest> for ReadStateService {
                 .wait_for_panics()
             }
 
+            // Used by the get_block (verbose) RPC and the StateService.
+            ReadRequest::BlockHeader(hash_or_height) => {
+                let state = self.clone();
+
+                tokio::task::spawn_blocking(move || {
+                    span.in_scope(move || {
+                        let header = state.non_finalized_state_receiver.with_watch_data(
+                            |non_finalized_state| {
+                                read::block_header(
```

Comment:
There's very similar code in some of these read state requests that we could probably make more generic.